src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Helpers/RemoveUnnecessaryImports/AbstractUnnecessaryImportsProvider.cs

```diff
@@ -18,8 +20,22 @@ internal abstract class AbstractUnnecessaryImportsProvider<TSyntaxNode> :
         public abstract ImmutableArray<TSyntaxNode> GetUnnecessaryImports(
             SemanticModel model, Func<SyntaxNode, bool>? predicate, CancellationToken cancellationToken);
 
-        public ImmutableArray<TSyntaxNode> GetUnnecessaryImports(SemanticModel model, CancellationToken cancellationToken)
-            => GetUnnecessaryImports(model, predicate: null, cancellationToken: cancellationToken);
+        public ImmutableArray<TSyntaxNode> GetUnnecessaryImports(SemanticModel model, TextSpan? span, CancellationToken cancellationToken)
+            => GetUnnecessaryImports(model, span, predicate: null, cancellationToken: cancellationToken);
+
+        public ImmutableArray<TSyntaxNode> GetUnnecessaryImports(
+            SemanticModel model, TextSpan? span, Func<SyntaxNode, bool>? predicate, CancellationToken cancellationToken)
+        {
+            if (span.HasValue)
+            {
+                // Bail out if there are no usings/imports in the filter span.
+                var node = model.SyntaxTree.GetNodeForSpan(span, findInTrivia: false, getInnermostNodeForTie: true, cancellationToken);
```

Comment:
surprised you needed getInnermostNodeForTie:true here.

Reply:
> surprised you needed getInnermostNodeForTie:true here.

It's probably me just being pedantic as I was not fully sure if there could be multiple nodes with the same span within a using node. You are correct though, it doesn't seem required, I'll remove it.